src/pattern-library/components/patterns/feedback/SpinnerPage.js

```diff
@@ -80,6 +102,25 @@ export default function SpinnerPage() {
           </Library.Demo>
         </Library.Example>
       </Library.Pattern>
+
+      <Library.Pattern title="SpinnerOverlay">
+        <p>
+          This component encapsulates a design pattern for a full-page loading
+          spinner.
+        </p>
+        <Library.Example title="Usage">
+          <Next.Usage componentName="SpinnerOverlay" />
+          <Library.Demo withSource>
+            <Button onClick={toggleOverlayOpen}>Show overlay</Button>
+            {overlayOpen && <SpinnerOverlay onClick={toggleOverlayOpen} />}
```

Comment:
An unfortunate effect of the conditional prop is that the `<SpinnerOverlay/>` component doesn't appear in the output when you view the "Source" page. Some possible ways to resolve this:

- Change the way the spinner overlay is shown so that it doesn't use a conditional. The downside is that the demo would not match exactly how the component would normally be used.
- Show the Demo and Source views at the same time, so the user can see the change in output when they click the button. This would still make it impossible to copy the source though
- Other ideas?

Longer term we could potentially change how the source view work so that conditionals, comments etc. are preserved.

Reply:
I can make `SpinnerOverlay`'s API align with `Overlay`'s by adding an `open` prop...

Reply:
Here's what I'm going to do for now:

<img width="841" alt="image" src="https://user-images.githubusercontent.com/439947/185405276-3f954d99-bd93-4fbc-a3c3-6e4458977204.png">

There are two forthcoming improvements:

1. After some changes down the line a bit are merged, the source will move _under_ the demo example. At this point in the stack of diffs, there's some logic in `Library` components that enforces that all `Demo`s are rendered "at the end" of examples. That will not be the case for long, allowing interleaving of Demos and other content.
2. I am considering extending the `Demo` component to take optional `source` content that would override its auto-generated source.